src/coreclr/src/tools/Common/TypeSystem/Ecma/EcmaModule.cs

```diff
@@ -90,7 +90,15 @@ protected override IEntityHandleObject CreateValueFromKey(EntityHandle handle)
                 switch (handle.Kind)
                 {
                     case HandleKind.TypeDefinition:
-                        item = new EcmaType(_module, (TypeDefinitionHandle)handle);
+                        var ecmaType = new EcmaType(_module, (TypeDefinitionHandle)handle);
+                        item = ecmaType;
+                        if (_module == _module.Context.CanonTypesModule)
```

Comment:
`_module.Context` is virtual property. It would be nice to fetch it just once into a local.